### ISSUE TITLE
cppcheck: common/test/run-features.c: error: Uninitialized variable (twice)

### ISSUE THREAD

**Issue by jsarenik** [ElementsProject/lightning#4167]
### Issue and Steps to Reproduce

Install `cppcheck`.

```
$ cppcheck --version
Cppcheck 2.2
$ #enter the C-lightning source tree
$ cd lightning
$ ./configure
...
$ make check-cppcheck
common/test/run-features.c:185:36: error: Uninitialized variable: f1 [uninitvar]
 for (size_t i = 0; i < ARRAY_SIZE(f1->bits); i++) {
                                   ^
common/test/run-features.c:217:36: error: Uninitialized variable: f [uninitvar]
 for (size_t i = 0; i < ARRAY_SIZE(f->bits); i++) {
                                   ^
make: *** [Makefile:462: check-cppcheck] Error 123
$ git rev-parse --short HEAD
ae4139ba1
```

**Comment by jsarenik** [comment]
Just a note. It fails the same way with `cppcheck 1.9` on Ubuntu 20.04

**Comment by jsarenik** [comment]
But the Ubuntu 18 on Travis has even older version of cppcheck which does not catch this.

**Comment by jsarenik** [comment]
Fixed by #4173
